### ISSUE TITLE
Duplicate fields

### ISSUE THREAD

**Issue by nulls** [saveourtool/cosv4k#1]
Some fields can be calculated or taken as is from OSV schema:
1. `severity[].level` can be calculated from `severity[].score`
2. `severity[].score_num` can be calculated from `severity[].score`
3. `contributors[].name` can be taken from `credits[].name`
4. `contributors[].email` can be taken from `credits[].contact[]`
5. `patches_detail[].patch_url` can be taken from `references[]` with `type = FIX`
6. `patches_detail[].issue_url` can be taken from `references[]` with `type = REPORT`
7. ~~`cwe_ids[]` can be calculated from `aliases[]` with prefix `CWE-`~~ cwe and cve are different: https://www.automox.com/blog/vulnerabilities-software-weaknesses-acronym-breakdown

**Comment by nulls** [comment]
@akuleshov7 please review

**Comment by akuleshov7** [comment]
@JustinB1eber

**Comment by nulls** [comment]
> 1 and 2, we added it for convenience of displaying so my suggestion is not to remove it.

Having a field which can be calculated from another leads to additional validation on our side:
if `severity[].level` is set, we should check that it's valid for provided `severity[].score` and the same for `severity[].score_num`.

Also, the JSON file is for communication between program (I don't think that human will read it), we can add additional calculated fields on UI only

**Comment by nulls** [comment]
> for 3, 4, using the credits[] filed is good makes it more simplified, and there is a contributors[].contributions field, if we replace 3 and 4 by credits[], we should add a credit[].contributions

Will move then extra fields from `contributors[]` to `credits[]`.

**Comment by nulls** [comment]
> for 5 and 6 my opinion is we should keep them inside the patched_detail[], so that this filed a more complete object.

Agree, taking into account that we want to move `patched_detail[]` to `range[]`
